frontend/libs/studio-components/src/components/StudioExpression/converters/complexToSimpleExpression.ts

```diff
@@ -77,12 +84,32 @@ const dataLookupFuncToSimpleFormat = ([source, key]: DataLookupFunc): SimpleSube
   }
 };
 
+const processActionToSimpleFormat = ([value]: DataLookupFunc): SimpleSubexpressionValue => {
+  if (Array.isArray(value) && value[0] !== SimpleSubexpressionValueType.GatewayActionContext) {
+    throw new Error(
```

Comment:
It's difficult to test that an error is thrown here because validations earlier in the code prevent us from reaching this point. This is included solely as a safety mechanism in case the validator is changed or fails to validate correctly. 😊 Therefore, this hasn't been tested, and I think it would be unreasonable to export the method just to write a test for it.